### PR TITLE
ElastiCache API support

### PR DESCRIPTION
**Pull request by bearleethere**
@benwilson512 @occurr 

Support for ElastiCache. Feedback is welcomed. We're currently using this for our production.

### CODE REVIEW

File: lib/ex_aws/elasticache.ex [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+  @version "2015-02-02"
+
+  @type tag :: {key :: atom, value :: binary}
+  @type node_group_configuration ::
+    {
+      primary_availability_zone :: binary,
+      replica_availability_zones :: [binary, ...],
+      replica_count :: integer,
+      slots :: binary
+    }
```
Comment (benwilson512) [inline lib/ex_aws/elasticache.ex]: I feel like a map might be a better choice here. It's slightly more verbose but it doesn't require folks remembering what order each of these things are in.

**Review by benwilson512 — COMMENTED** [review]
This looks great, I really appreciate the extensive docs.

**Comment by bearleethere** [comment]
Good point. I will make the change when I get home.

File: lib/ex_aws/elasticache.ex [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+          "NodeGroupConfigurations.NodeGroupConfiguration.2.ReplicaAvailabilityZones.AvailabilityZone.1" => "us-east-1a",
+          "NodeGroupConfigurations.NodeGroupConfiguration.2.ReplicaAvailabilityZones.AvailabilityZone.2" => "us-east-1c",
+          "NodeGroupConfigurations.NodeGroupConfiguration.2.ReplicaCount" => 2,
+          "NodeGroupConfigurations.NodeGroupConfiguration.2.Slots" => 2,
+          "Version" => "2015-02-02"
+        },
+        path: "/",
+        service: :elasticache}
+  """
+  @type node_group_configuration ::
```
Comment (bearleethere) [inline lib/ex_aws/elasticache.ex]: @benwilson512 @occurr 
Converted it to a map. 

The new version looks a lot clearer:

```
ExAws.ElastiCache.create_replication_group("myRepGroup", "My Rep Group",
  [node_group_configurations: [
     %{primary_availability_zone: "us-east-1a",
         replica_availability_zones: ["us-east-1b", "us-east-1c"],
         replica_count: 2,
         slots: 0},
     %{primary_availability_zone: "us-east-1b",
         replica_availability_zones: ["us-east-1a", "us-east-1c"],
         replica_count: 2,
         slots: 2}
]])
```

File: lib/ex_aws/elasticache.ex [hunk 3]
```diff
@@ -194,13 +195,12 @@ defmodule ExAws.ElastiCache do
         path: "/",
         service: :elasticache}
   """
-  @type node_group_configuration ::
-    %{
+  @type node_group_configuration :: [
```
Comment (bearleethere) [inline lib/ex_aws/elasticache.ex]: Switched to a keyword list to keep it consistent with all of the other elixir AWS libraries.

**Comment by bearleethere** [comment]
@benwilson512, did you have a chance to look at the revisions?

**Comment by bearleethere** [comment]
@benwilson512, just want to follow up if you had a chance to look at the revisions. I would like to merge this with master.

**Pull request merged**